src/pip/_internal/commands/cache.py

```diff
@@ -59,8 +59,9 @@ def run(self, options, args):
 
         # Determine action
         if not args or args[0] not in handlers:
-            logger.error("Need an action ({}) to perform.".format(
-                ", ".join(sorted(handlers)))
+            logger.error(
+                "Need an action ({}) to perform.",
```

Comment:
It looks like this needs to be made to use `%s`, since `.format` was removed.

Reply:
Thanks for the catch, I've just integrated this into one of the commits.